bin/pycbc_make_skymap

```diff
@@ -91,41 +93,47 @@ def main(trig_time, mass1, mass2, spin1z, spin2z, f_low, f_upper, ifar, ifos, th
 
     # parameters to fit a single-template inspiral job nicely
     # around the trigger time, without requiring too much data
-    segment_start_pad = 144
-    segment_end_pad = 16
-    segment_length = 256
-    gps_start_time = rough_time - segment_start_pad - (segment_length - segment_start_pad - segment_end_pad) // 2
-    gps_end_time = gps_start_time + segment_length
-    psd_segment_length = 16
-    psd_segment_stride = psd_segment_length / 2
-    psd_num_segments = 31
+    
+    # Padding set by 16 * 2 for psd and buffer for other filtering
+    pad = 50
+    template_duration = spa_length_in_time(mass1=mass1, mass2=mass2, 
+                                           f_lower=f_low, phase_order=-1)
+    segment_length = int(nearest_larger_binary_number(template_duration + pad))
+    # set minimum so there is enough for a psd estimate
+    if segment_length < 128:
+        segment_length = 128
+    logging.info('Using segment length: %s', segment_length) 
+    
+    gps_end_time = int(rough_time + pad // 2)
+    gps_start_time = gps_end_time - segment_length
+    logging.info("Using data: %s-%s", gps_start_time, gps_end_time)
+    
+    highpass_frequency = int(f_low * 0.7)
+    logging.info("Setting highpass: %s Hz", highpass_frequency)
 
     for i, ifo in enumerate(ifos):
         followup_data[ifo] = {}
         command = ["pycbc_single_template",
         "--verbose",
         "--segment-length", str(segment_length),
-        "--segment-start-pad", str(segment_start_pad),
-        "--segment-end-pad", str(segment_end_pad),
+        "--segment-start-pad", "0",
+        "--segment-end-pad", "0",
         "--psd-estimation","median",
-        "--psd-segment-length", str(psd_segment_length),
-        "--psd-segment-stride", str(psd_segment_stride),
-        "--psd-inverse-length", str(psd_segment_length),
-        "--approximant",'SPAtmplt:mtotal<4','SEOBNRv4_ROM:else',
+        "--psd-segment-length", "16",
+        "--psd-segment-stride", "8",
+        "--psd-inverse-length", "16",
+        "--approximant", approximant,
         "--order","-1",
-        "--psd-num-segments", str(psd_num_segments),
         "--taper-data","1",
         "--allow-zero-padding",
         "--autogating-threshold","100",
         "--autogating-cluster","0.5",
         "--autogating-width","0.25",
         "--autogating-taper","0.25",
         "--autogating-pad","16",
-        "--minimum-chisq-bins","3",
-        "--strain-high-pass","15.",
-        "--pad-data","8",
-        "--template-start-frequency","27",
-        "--chisq-bins",'0.72*get_freq("fSEOBNRv4Peak",params.mass1,params.mass2,params.spin1z,params.spin2z)**0.7']
+        "--strain-high-pass", str(highpass_frequency),
+        "--pad-data","0",
+        "--chisq-bins",'16']
```

Comment:
I think one of the original ideas behind this script was to generate a GraceDB upload as close as possible to how a search would upload it. So I would prefer to keep the expression for the chisq-bins as it was.

Reply:
@titodalcanton Ok, though note that the other options weren't exactly the same as the search.